tests/scripts/components.sh

```diff
@@ -0,0 +1,4349 @@
+#! /usr/bin/env bash
+
+# components.sh
```

Comment:
We're going to want to split components into multiple files at one point. I suggest we start now, since we're moving the code around.

For the project split, we're going to have crypto-only components, TLS-only components, and components that should exist in both (but not calling the TLS tests in the crypto repository).

Also, I have reasons to separate tests into several categories and I'd like to facilitate analyzing and running them independently. Most component go into the category of configuration testing. There's also a category of compiler testing (`build_armcc`, `test_*_opt`, …), non-build checks (`check_*`), extra sanitizers (msan, valgrind), platform testing (`test_m32_*`, …), build system testing (various `test_cmake_*`). Since we're moving code into new files anyway, I'd like to directly create several separate files.